tests/e2e/lib/virt_helpers.go

```diff
@@ -0,0 +1,408 @@
+// health status field is "healthy". Uses dynamic client to avoid uprooting lots
+// of package dependencies, which should probably be fixed later.
+func (v *VirtOperator) checkHco() bool {
+	resourceId := schema.GroupVersionResource{
+		Group:    "hco.kubevirt.io",
+		Resource: "hyperconvergeds",
+		Version:  "v1beta1",
+	}
+
+	unstructuredHco, err := v.Dynamic.Resource(resourceId).Namespace(v.Namespace).Get(context.Background(), "kubevirt-hyperconverged", v1.GetOptions{})
+	if err != nil {
+		log.Printf("Error getting HCO: %v", err)
+		return false
+	}
+
+	health, ok, err := unstructured.NestedString(unstructuredHco.UnstructuredContent(), "status", "systemHealthStatus")
+	if err != nil {
+		log.Printf("Error getting HCO health: %v", err)
+		return false
+	}
+	if !ok {
+		log.Printf("HCO health field not populated yet")
+		return false
+	}
+	log.Printf("HCO health status is: %s", health)
+
+	return health == "healthy"
+}
+
+// Creates the target virtualization namespace, likely openshift-cnv or kubevirt-hyperconverged
+func (v *VirtOperator) installNamespace() error {
+	err := v.Client.Create(context.Background(), &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: v.Namespace}})
+	if err != nil {
+		log.Printf("Failed to create namespace %s: %v", v.Namespace, err)
+		return err
+	}
+	return nil
+}
+
+// Creates the virtualization operator group
+func (v *VirtOperator) installOperatorGroup() error {
+	group := &operatorsv1.OperatorGroup{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "kubevirt-hyperconverged-group",
+			Namespace: v.Namespace,
+		},
+		Spec: operatorsv1.OperatorGroupSpec{
+			TargetNamespaces: []string{
+				v.Namespace,
+			},
+		},
+	}
+	err := v.Client.Create(context.Background(), group)
+	if err != nil {
+		if !strings.Contains(err.Error(), "already exists") {
+			log.Printf("Failed to create operator group: %v", err)
+			return err
+		}
+	}
+	return nil
+}
+
+// Creates the subscription, which triggers creation of the ClusterServiceVersion.
+func (v *VirtOperator) installSubscription() error {
+	subscription := &operatorsv1alpha1.Subscription{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "hco-operatorhub",
+			Namespace: v.Namespace,
+		},
+		Spec: &operatorsv1alpha1.SubscriptionSpec{
+			CatalogSource:          "redhat-operators",
+			CatalogSourceNamespace: "openshift-marketplace",
+			Package:                "kubevirt-hyperconverged",
+			Channel:                "stable",
+			StartingCSV:            v.Csv,
+			InstallPlanApproval:    operatorsv1alpha1.ApprovalAutomatic,
+		},
+	}
+	err := v.Client.Create(context.Background(), subscription)
+	if err != nil {
+		log.Printf("Failed to create subscription: %v", err)
+		return err
+	}
+
+	return nil
+}
+
+// Creates a HyperConverged Operator instance. Another dynamic client to avoid
+// bringing in the KubeVirt APIs for now.
+func (v *VirtOperator) installHco() error {
+	resourceId := schema.GroupVersionResource{
```

Comment:
Could this schema be defined outside of the code or at least somewhere common together with other schema attributes which are not really functional.

Reply:
I'm not totally sure what you are asking for here, but I moved the GVRs together to the top of the file.